ethsigner/commandline/src/test/java/tech/pegasys/ethsigner/CommandlineParserTest.java

```diff
@@ -84,30 +89,42 @@ public void fullyPopulatedCommandLineParsesIntoVariables() throws UnknownHostExc
   }
 
   @Test
-  public void helpMessageIsShown() {
-    parseCommand("--help");
-    final String expectedOutputStart = String.format("Usage:%n%nethsigner [OPTIONS]");
-    assertThat(commandOutput.toString()).startsWith(expectedOutputStart);
+  public void mainCommandHelpIsDisplayedWhenNoOptionsOtherThanHelp() {
+    final boolean result = parser.parseCommandLine("--help");
+    assertThat(result).isTrue();
+    assertThat(commandOutput.toString()).isEqualTo(defaultUsageText);
+  }
+
+  @Test
+  public void mainCommandHelpIsDisplayedWhenNoOptionsOtherThanHelpWithoutDashes() {
+    final boolean result = parser.parseCommandLine("help");
+    assertThat(result).isTrue();
+    assertThat(commandOutput.toString()).containsOnlyOnce(defaultUsageText);
+  }
+
+  @Test
+  public void reverseHelpRequestShowsSubCommandHelp() {
+    final boolean result = parser.parseCommandLine("help", subCommand.getCommandName());
+    assertThat(result).isTrue();
+    assertThat(commandOutput.toString()).isEqualTo(nullCommandHelp);
   }
 
   @Test
   public void missingSubCommandShowsErrorAndUsageText() {
-    parseCommand(validCommandLine());
-    assertThat(commandOutput.toString()).startsWith(CommandlineParser.MISSING_SUBCOMMAND_ERROR);
-    final String expectedOutputStart = String.format("Usage:%n%nethsigner [OPTIONS]");
-    assertThat(commandOutput.toString()).contains(expectedOutputStart);
+    final boolean result = parser.parseCommandLine(validCommandLine().split(" "));
```

Comment:
just looks a bit funny with "validCommandLine" and yousay it is supposed to fail ...

Reply:
true ... hmmm ... renamed (better?)